actionmailbox/lib/action_mailbox/errors.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require "active_support/actionable_error"
+
+module ActionMailbox
+  # Generic base class for all Action Mailbox exceptions.
+  class Error < StandardError; end
+
+  # Raised when we detect that Action Mailbox has not been initialized.
+  class SetupError < Error
+    include ActiveSupport::ActionableError
+
+    def initialize(message = nil)
+      super(message || <<~MESSAGE)
+        Action Mailbox does not appear to be installed. Do you want to
+        install it now?
+      MESSAGE
+    end
+
+    trigger ActiveRecord::StatementInvalid do |error|
```

Comment:
This is the new interface. It says:

> Trigger the actionable error if an `ActiveRecord::StatementInvalid` is detected and it matches the condition given in the block, e.g. it's about the missing `action_mailbox_inbound_emails` table.

Reply:
With this interface, though, the error class should not be autoloaded, since it can only register itself to the trigger list, if it was loaded. But if the error class is autoloaded, it will never get a chance to do that. Have you thought about a way around that?